src/LanguageServer/Impl/Indexing/MostRecentSymbols.cs

```diff
@@ -30,12 +32,11 @@ class MostRecentDocumentSymbols : IDisposable {
         }
 
         private async Task ParseAsync(CancellationToken cancellationToken) {
-            var ast = await _indexParser.ParseAsync(_path, cancellationToken);
```

Comment:
If this throws, whoever waits on TCS will hang since `SetFileTcsResult` will never get called. Proper way is
```cs
try {
do stuff
    _tcs.TrySetResult(...)
} catch(OperationCanceledException) {
    _tcs.TrySetCanceled();
} catch(Exception ex) when !(ex.IsCriticalException()) {
   _tcs.TrySetException(ex)
   throw;
}
```